src/mame/drivers/tetrisp2.cpp

```diff
@@ -1860,53 +1839,174 @@ TIMER_DEVICE_CALLBACK_MEMBER(stepstag_state::field_cb)
+		}
+	}
+	else if((m_stepstag_main2pc & 0x0fff)==0x715)	//sound effects 2
+	{
+		position = 0x400000;
+
+		inFile=fopen("vjs02.vjs", "rb");
+		if( inFile )
+		{
+			fseek(inFile, 0, SEEK_END);
+			size = ftell(inFile);
+			rewind(inFile);
+			if(size > total_ram - position)
+				size = total_ram - position;
+			fread(m_ymzram + position, 1, size, inFile);
+			fclose(inFile);
+			memcpy(m_ymzram1 + position, m_ymzram + position, size);
+		}
+	}
+	else if((m_stepstag_main2pc & 0x0f00)==0x7ff)
+	{
+	}
+	else if((m_stepstag_main2pc & 0x0f00)==0x700)	 printf(" cmd 0x%x !! unknown !!!\n", m_stepstag_main2pc);
+
+	m_stepstag_main2pc = 0;
+}
+
+/*void stepstag_state::simulate_vj_sub()
+{
+}*/
+
 void stepstag_state::stepstag(machine_config &config)
 {
-	M68000(config, m_maincpu, XTAL(12'000'000)); // unknown
+	M68000(config, m_maincpu, XTAL(12'000'000));	// unknown
 	m_maincpu->set_addrmap(AS_PROGRAM, &stepstag_state::stepstag_map);
 
-	constexpr XTAL subxtal = XTAL(42'954'545); // unknown
-	constexpr XTAL sub_pixel_clock = subxtal/6;
+	constexpr XTAL subxtal = XTAL(13'500'000);	// 2x ntsc mpeg-1 clock
+	constexpr XTAL sub_pixel_clock = subxtal/2;
 
-	M68000(config, m_subcpu, subxtal/3);
+	M68000(config, m_subcpu, subxtal);
 	m_subcpu->set_addrmap(AS_PROGRAM, &stepstag_state::stepstag_sub_map);
-	TIMER(config, "field_timer").configure_periodic(FUNC(stepstag_state::field_cb), attotime::from_hz(30));
+	TIMER(config, "field_timer").configure_periodic(FUNC(stepstag_state::field_cb), attotime::from_hz(29.97));	//standard NTSC 525 interlaced mode
 
 	NVRAM(config, "nvram", nvram_device::DEFAULT_ALL_0);
 
 	WATCHDOG_TIMER(config, "watchdog");
 
 	// video hardware
 
-	// this screen arrangement is weird:
-	// it writes a regular 320x224 screen setup to the CRTC but none of these matches a 352 width,
-	// we are either missing a bit from the config regs or those writes are null and
-	// these screens are driven by something else.
-	// Also note: main 68k tilemap/sprite/palette aren't even displayed with this arrangement,
-	// even tho usage is minimal (POST/test mode), maybe just a left-over ...
-	screen_device &lscreen(SCREEN(config, "lscreen", SCREEN_TYPE_RASTER));
-	lscreen.set_orientation(ROT270);
-	setup_non_sysctrl_screen(config, &lscreen, subxtal);
-	lscreen.set_screen_update(FUNC(stepstag_state::screen_update_stepstag_left));
-
+	// The mid-CRT is supposed to have a video switcher to choose one from two sources:
+	// In normal game operation mode, a 352*240*29.97Hz video from the sub cpu board;
+	// While in POST/test mode, a 320x224*60.6Hz from the main cpu board, similar to Rock'n Tread series
+	// TODO: Where is the switching signal from, sysctrl?
 	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
 	m_screen->set_orientation(ROT0);
 	// TODO: connected to the non sysctrl CRTC anyway?
 	m_screen->set_raw(XTAL(48'000'000)/8, 384, 0, 320, 263, 0, 224);
+///	m_screen->set_raw(sub_pixel_clock, 429, 0, 352, 525, 0, 240);	// should be 0-479 interlaced
```

Comment:
Triple dash at start, also again not supposed to change up the clock directly but instead superimpose.

Reply:
In game mode, video signal is output from the sub-cpu-board, while the sysctrl only manages video hardware on the main-cpu-board.

Reply:
Well they are two different signals, and one genlocks over the other. It doesn't change source timing _signals_.

Reply:
https://www.youtube.com/watch?v=Oqb7dDyD4fM

In this footage, some switching noise can be observed at the beginning;
Also, between the booting (text displaying) period and in-game period (3:15), the speed of the up-scrolling dark bar is obvious different. It means the two scan rates is different.
So, the two video signal source, each one has its own timing, and one doesn't genlock another. .